### PR TITLE
Add volumetric instance segmentation scripts

### PR DESCRIPTION
**Pull request by anwai98**

### CODE REVIEW

File: scripts/volumetric_evaluation/evaluate_lucchi.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+            stop_upper=False, iou_threshold=iou_threshold, projection=projection, box_extension=box_extension
+        )
+
+        final_segmentation[this_seg == 1] = label_id
+
+    msa = mean_segmentation_accuracy(final_segmentation, instance_labels)
+    return msa
+
+
+def main(args):
```
[the remark this answers is not in the document]
Comment (constantinpape) [reply scripts/volumetric_evaluation/evaluate_lucchi.py:86]: And then use the function here to run the grid search for lucchi.

File: scripts/volumetric_evaluation/evaluate_lucchi.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    predictor = _get_model(model_type=model_type, checkpoint_path=checkpoint)
+
+    test_volume_path = os.path.join(input_path, "lucchi_test.h5")
+    volume, labels = get_raw_and_label_volumes(test_volume_path)
+
+    # precompute embeddings
+    image_embeddings = util.precompute_image_embeddings(predictor, volume, save_path=embedding_path, ndim=3)
+
+    # let's get the instances from the label volume
+    instance_labels = label(labels)
```
Comment (constantinpape) [inline scripts/volumetric_evaluation/evaluate_lucchi.py:46]: We should assume that we get instance labels passed to the function.
(So for lucchi you can do this beforehand)

**Review by constantinpape — CHANGES_REQUESTED** [review]
The implementation looks good. But we should refactor it into `micro_sam/evaluation`, so that it can be reused for evaluating other 3d segmentation problems.

File: scripts/volumetric_evaluation/evaluate_lucchi.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (anwai98) [reply scripts/volumetric_evaluation/evaluate_lucchi.py:86]: Okie, makes sense. I will create a set of default grid search params (the same as for AMG / AIS) (and also leave the possibility for the users to overwrite with their choice of parameters)

[the remark this answers is not in the document]
Comment (anwai98) [reply scripts/volumetric_evaluation/evaluate_lucchi.py:86]: Done! ;)

File: micro_sam/multi_dimensional_segmentation.py [hunk 3]
```diff
@@ -29,6 +29,7 @@ def segment_mask_in_volume(
     projection: Union[str, dict],
     progress_bar: Optional[Any] = None,
     box_extension: float = 0.0,
+    verbose: bool = True
```
Comment (constantinpape) [inline micro_sam/multi_dimensional_segmentation.py:32]: I would set this to `False` by default. Otherwise we will get a lot of output in the plugin.

**Review by constantinpape — COMMENTED** [review]
Looks good now! Just one minor point.

File: micro_sam/multi_dimensional_segmentation.py (hunk 3 above)
Comment (anwai98) [reply micro_sam/multi_dimensional_segmentation.py:32]: Done! ;)

**Review by constantinpape — APPROVED** [review]

**Pull request merged**